src/main/java/com/google/devtools/build/lib/remote/Retrier.java

```diff
@@ -141,6 +144,8 @@ public void recordSuccess() {}
 
   public static class FailureRateCircuitBreaker implements CircuitBreaker {
 
+    private static Logger logger = Logger.getLogger(FailureRateCircuitBreaker.class.getName());
+
     // 100 arbitrarily chosen as a fair balance between giving up
     // too quickly vs trying too much and giving up too late.
     public static final int DEFAULT_MIN_EXECUTIONS_TO_COMPUTE_FAILURE_RATE = 100;
```

Comment:
I'd say this is not needed if we measure the failure rate based on a time window.
This will compete with the `remote_failure_rate_window_size` option; if a very (too) small time window is used and the number of executions is too small to trigger failure rate calculation, it will cause confusions and will mask the real issue (chosen time window being too small).
So I think we can get rid of it now and also simplify the code a bit.

Reply:
Agree that having two indicators could be confusing. My concern is that without a minimum number, if the first request fails, for any reason, we'd stop communicating with the remote immediately.

Reply:
Yeah, I get that. In that case, wouldn't it mean that the chosen `remote_failure_rate_window_size` value is too small? As an alternative to keeping the `DEFAULT_MIN_EXECUTIONS_` value, we could constrain `remote_failure_rate_window_size` to allow only for reasonable values, e.g. >60s.

Reply:
So you're suggesting to use the time window as a minimum period of time before computing the failure rate?

Reply:
Yes, to not allow unreasonably small time windows. To force it be >30 seconds, for example.

Reply:
I think the simplest approach here would be:
- compute failure rate only if `remote_max_failure_rate` > 0
- enforce a reasonable time window in which the failure rate is calculated, e.g. >30s, default to 5 mins
- stop comms with the cache if there are >0 requests and the computed failure rate for that time window exceeds `remote_max_failure_rate`